### PR TITLE
fuzz: add more coverage for `ScriptPubKeyMan`

### PR DESCRIPTION
**Pull request by brunoerg**
This PR adds more coverage for `ScriptPubKeyMan`:

- Check `GetKey` and `HasPrivKey` after adding descriptor key.
- Cover `GetEndRange` and `GetKeyPoolSize`.
- Cover `MarkUnusedAddresses` with the scripts from ScriptPubKeys and `GetMetadata` with the destinations from them.

### CODE REVIEW

File: src/wallet/test/fuzz/scriptpubkeyman.cpp [hunk 1]
```diff
@@ -2,38 +2,17 @@
...
-#include <wallet/types.h>
 #include <wallet/wallet.h>
 #include <wallet/walletutil.h>
 
-#include <map>
-#include <memory>
-#include <optional>
-#include <string>
-#include <utility>
-#include <variant>
```
[the remark this answers is not in the document]
Comment (brunoerg) [reply src/wallet/test/fuzz/scriptpubkeyman.cpp]: Nevermind, just dropped this change.

**Review by marcofleon — COMMENTED** [review]
I generated a coverage report for the current `ScriptPubKeyMan` harness and for the updated harness in this PR.

Before:
Screenshot 2024-05-19 at 6 18 27 AM
Screenshot 2024-05-19 at 6 18 37 AM

After:
Screenshot 2024-05-19 at 7 06 41 AM
Screenshot 2024-05-19 at 7 06 54 AM

So more coverage in `scriptpubkeyman.cpp` but it seems less total coverage when running the indivdual harness. My question here is do the totals matter when fuzzing a single target. Because there are probably other harnesses that would hit those other files (other than `scriptpubkeyman.cpp`). 

Basically, does total coverage only matter when generating a report from fuzzing all the targets?

**Comment by brunoerg** [comment]
> I generated a coverage report for the current ScriptPubKeyMan harness and for the updated harness in this PR.

@marcofleon It's good to mention how you run it. How many hours? From seed corpus (note that changes can invalidate it)?

**Comment by marcofleon** [comment]
Got it, thanks @brunoerg. I'll redo and get back to you then. I also realized I was looking at the wrong line in the coverage report. I should probably be looking at `wallet/scriptpubkeyman.cpp` not `wallet/test/fuzz/scriptpubkeyman.cpp`.

**Review by marcofleon — COMMENTED** [review]
Tested ACK e3249f21111f1dd4beb66f10af933c34a36c30ac. I ran the updated harness for ~9 hours on an empty corpus, generated a coverage report, and checked that the new functions mentioned were hit. Coverage of `scriptpubkeyman.cpp` increased.

**Review by murchandamus — COMMENTED** [review]
Tested ACK e3249f21111f1dd4beb66f10af933c34a36c30ac

I also fuzzed a bit: I ran a recent commit of the `master` branch and this PR’s e3249f21111f1dd4beb66f10af933c34a36c30ac against `qa-assets:HEAD` and then again after fuzzing about 12 CPU hours on top of `qa-assets:HEAD`.

|   | `qa-assets` | `qa-assets` + fuzzing |
| --- | ------------- | ------------- |
| `master` | cov: 16525 ft: 96185  | cov: 16874 ft: 104431 |
| #30134: e3249f21111f1dd4beb66f10af933c34a36c30ac | cov: 17095 ft: 83926  | cov: 17699 ft: 98966 |

This PR increases the coverage with the existing seeds and further improves even after light fuzzing.
